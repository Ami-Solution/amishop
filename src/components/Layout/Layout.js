import styled, { css } from 'styled-components';

const Layout = styled.div`
  display: flex;
  flex-flow: column nowrap;
  align-self: stretch;
  min-height: 100%;
  width: 100%;
  text-align: center;
  position: relative;

  ${({ textAlignLeft }) =>
    textAlignLeft &&
    css`
      text-align: left;
    `};

  ${({ isFullScreen }) =>
    isFullScreen &&
    css`
      overflow: hidden;
      height: 100%;
      max-height: 100%;
    `};
`;

Layout.HeaderLarge = styled.header`
  flex: 0 0 13vh;
  display: flex;
  flex-flow: column;
  justify-content: flex-end;
`;

Layout.Header = styled.header`
  display: block;
`;

Layout.Body = styled.div`
  flex: 1;
  display: flex;
  flex-flow: column nowrap;
  justify-content: space-around;
`;

Layout.ScrollableBody = styled.div`
  flex: 1;
  overflow-y: auto;
  display: flex;
  flex-flow: row;
  align-items: center;
  justify-content: space-around;
`;

Layout.Footer = styled.footer`
  flex: 0;
`;

/* allow storybook to display correctly the name */

Layout.displayName = 'Layout';
Layout.Header.displayName = 'Layout.Header';
Layout.HeaderLarge.displayName = 'Layout.HeaderLarge';
Layout.Body.displayName = 'Layout.Body';
Layout.ScrollableBody.displayName = 'Layout.ScrollableBody';
Layout.Footer.displayName = 'Layout.Footer';

export default Layout;
