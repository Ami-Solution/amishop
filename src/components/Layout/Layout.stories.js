/* eslint-disable */
import React from 'react'
import styled, { css } from 'styled-components'
import { storiesOf, addDecorator } from '@storybook/react'
import { withInfo } from '@storybook/addon-info'
import Layout from './index'

storiesOf('03 - Components', module)
  .addDecorator(story => <div style={{ margin: '-30px auto', maxWidth: '320px' }}>{story()}</div>)
  .add(
    'Layout',
    withInfo()(() => (
      <Layout>
        <Layout.Header>
          <div style={{ padding: '10px 15px' }}>
            <b>header:</b> verticaly centered inside a block that use 20% of the viewport
          </div>
        </Layout.Header>
        <Layout.Body>
          <div style={{ padding: '10px 15px' }}>
            <b>body:</b> verticaly centered inside the available height <br />
            <br />
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolorem facere, reiciendis
            repudiandae tenetur animi corrupti quam itaque ducimus illo, rerum illum aliquid quae,
            cumque inventore dolorum sint fugiat asperiores. Rerum! Lorem ipsum dolor sit amet,
            consectetur adipisicing elit.<br />
            <br />
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Dolorem facere, reiciendis
            repudiandae tenetur animi corrupti quam itaque ducimus illo, rerum illum aliquid quae,
            cumque inventore dolorum sint fugiat asperiores. Rerum! Lorem ipsum dolor sit amet,
            consectetur adipisicing elit.
          </div>
        </Layout.Body>
        <Layout.Footer>
          <div style={{ padding: '10px 15px' }}>
            <b>footer:</b> stick to bottom
          </div>
        </Layout.Footer>
      </Layout>
    ))
  )
