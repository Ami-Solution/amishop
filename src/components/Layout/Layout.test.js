/* global describe it expect */
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Layout from './index';

describe('Component Layout', () => {
  it('should render Layout', () => {
    const component = renderer.create(
      <Layout>
        <Layout.Header>Header</Layout.Header>
        <Layout.Body>Body</Layout.Body>
        <Layout.Footer>Footer</Layout.Footer>
      </Layout>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
