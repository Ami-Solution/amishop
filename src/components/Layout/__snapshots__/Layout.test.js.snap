// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Layout should render Layout 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  min-height: 100%;
  width: 100%;
  text-align: center;
  position: relative;
}

.c1 {
  display: block;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c3 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
}

<div
  className="c0"
>
  <header
    className="c1"
  >
    Header
  </header>
  <div
    className="c2"
  >
    Body
  </div>
  <footer
    className="c3"
  >
    Footer
  </footer>
</div>
`;
